firmware/quadruna/VC/src/io/io_imu.c

```diff
@@ -0,0 +1,73 @@
+#include "io_imu.h"
+#include "hw_i2c.h"
+#include "app_canTx.h"
+#include <math.h>
+#include <stdio.h>
+
+extern I2C_HandleTypeDef hi2c2;
+static I2cInterface     *imu;
+
+// Default sensitivity for LSM6DSM is 0.061 mg/digit
+const float SENSITIVITY = 0.061f;
+
+bool io_imu_init()
+{
+    imu->i2c_handle     = &hi2c2;
+    imu->target_address = 0x6B;
+    imu->timeout_ms     = 100;
+
+    return hw_i2c_is_target_ready(imu);
+}
+
+bool io_imu_init_accelerometer()
+{
+    uint8_t buffer[1];
+    buffer[0] = 0x40; // turn on accelerometer to normal mode
+
+    return hw_i2c_mem_write(imu, 0x10, 1, buffer, sizeof(buffer));
+}
+
+void read_register(uint8_t *rx_buffer, uint16_t mem_addr)
+{
+    hw_i2c_mem_read(imu, mem_addr, 1, &rx_buffer[0], 1);         // LSB
+    hw_i2c_mem_read(imu, (mem_addr + 0x1), 1, &rx_buffer[1], 1); // MSB
+}
```

Comment:
I believe you can just invoke `hw_i2c_mem_read` once with a size of 2 bytes, and it'll do the same as you're doing here.